pkg/awsvpc/sg_analysis.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package awsvpc
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/ipblock"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/commonvpc"
+)
+
+type AWSSGAnalyzer struct {
+	sgResource         *types.SecurityGroup
+	referencedIPblocks []*ipblock.IPBlock
+	sgMap              map[string]*commonvpc.SecurityGroup
+}
+
+func NewAWSSGAnalyzer(sg *types.SecurityGroup) *AWSSGAnalyzer {
+	res := &AWSSGAnalyzer{sgResource: sg}
+	return res
+}
+
+func (sga *AWSSGAnalyzer) Name() *string {
+	return sga.sgResource.GroupName
+}
+
+func (sga *AWSSGAnalyzer) getRemoteCidr(ipRanges []types.IpRange, userIDGroupPairs []types.UserIdGroupPair) (
+	remote *ipblock.IPBlock, err error) {
+	remote = ipblock.New()
+	for i := range ipRanges {
+		target, _, err := commonvpc.GetIPBlockResult(ipRanges[i].CidrIp, nil, nil, sga.sgMap)
```

Comment:
the aws documentation for `CidrIP` says:
```
// The IPv4 address range. You can either specify a CIDR block or a source
// security group, not both. To specify a single IPv4 address, use the /32 prefix
// length.
```
does this mean it can also specify a SG ? not just ipv4 cidr?